src/index.ts

```diff
@@ -158,7 +158,10 @@ async function run(): Promise<void> {
 
     const registryWithoutTrailingSlash = registry.replace(/\/$/, "");
 
-    const creds = `${username}:${password}`;
+    let creds = "";
+    if (username && password) {
```

Comment:
I am wondering if we need to handle the case here when user has not provided only `username` but not `password` or vice versa.
I this case we can log a message (may be a warning message) to say something like:
`username is provided but password is missing`

Reply:
Yes, that's worth adding.